Feature/sensor updates

tpmanley opened the pull request:
Please review @workingmonk @umangparekh

tpmanley committed:
Fix whitespace issues - no code changes

Replaced spaces with tabs for indentation and removed some unnecessary
white space.

tpmanley commented:
This PR includes a whitespace change. It was just getting to be too difficult to make changes because of all the inconsistent whitespace issues. To see the non-whitespace changes just add `?w=1` to the end of the PR URL.

tpmanley commented:
I found a bug in the new battery code for multi/moition/moisture that I need to fix and will update the PR.

workingmonk commented on devicetypes/smartthings/smartsense-multi-sensor.src/smartsense-multi-sensor.groovy:
```diff
@@ -302,132 +302,141 @@ def getTemperature(value) {
+			if (device.getDataValue("manufacturer") == "SmartThings") {
+				volts = rawValue // For the batteryMap to work the key needs to be an int
+				def batteryMap = [28:100, 27:100, 26:100, 25:90, 24:75, 23:50,
+								  22:30, 21:15, 20:5, 19:3, 18:0, 17:0, 16:0]
```
centralite sensors tend to go bust at 23

tpmanley replied:
Yeah, this doesn't apply to CL. Check out the Jira for more info

workingmonk replied:
oops, overlooked your manufacturer check there

tpmanley commented on devicetypes/smartthings/smartsense-moisture-sensor.src/smartsense-moisture-sensor.groovy:
```diff
@@ -215,24 +215,47 @@ def getTemperature(value) {
-		result.descriptionText = "${linkText} battery has too much power (${volts} volts)."
-	}
+
+	if (rawValue == 255) {}
```
Previously this sensor wasn't checking to make sure the battery attribute was valid (!= 0xFF)

tpmanley commented on devicetypes/smartthings/smartsense-motion-sensor.src/smartsense-motion-sensor.groovy:
```diff
@@ -230,30 +230,46 @@ def getTemperature(value) {
-	def descriptionText
 
-	if (rawValue == 0) {}
+	if (rawValue == 255) {}
```
Technically 0 is a valid value, but if it is actually 0 then it's wrong because the sensor couldn't possibly be powered. I think I'm going to update all the sensors to report '--' if rawValue is 0 or 255

umangparekh commented:
Pardon my ignorance about the DTH framework. But, Is there a way all the sensors could be derived from a common base class? If this can be done, then I suggest we move the batteryMap change for SmartThings sensors in the base class. Let me know what you guys think. 

Other than that: :+1:

tpmanley committed:
Use table for battery voltage to percent remaining calculation

The new table based approach yields a more accurate battery percentage
remaining than the old linear calculation.

Resolves:
    https://smartthings.atlassian.net/browse/SMJN-39

tpmanley commented:
@umangparekh , there isn't really a good way to do that unfortunately

workingmonk commented on devicetypes/smartthings/arrival-sensor-ha.src/arrival-sensor-ha.groovy:
```diff
@@ -88,27 +88,33 @@ private handleReportAttributeMessage(String description) {
+private handleBatteryEvent(volts) {
     def eventMap = [
         name: 'battery',
         value: '--'
```
not sure if this is a good idea. I have seen devices sometimes inadvertently report 0 or FF in the middle of their life cycle. Setting it to '--' causes two issues:
1. In the device tile you suddenly lose the currently battery level of the device. And probably won't get an update on its battery for probably a day
2. Battery level is tied to Low Battery Monitoring, not sure how well it is going to accept the non-int value

tpmanley replied:
I did it that way because that's the way the other sensors were implemented. Do you think I should change them all to not report any battery event if the value is 0 or 255?

workingmonk replied:
That might make more sense. This along with the map work that you've done for battery is something that we should standardize for other battery devices

tpmanley commented:
@workingmonk, I fixed the issue for the arrival. I think it's a bigger change to the other sensors which I'm not sure should be made as part of this PR. What do you think?

workingmonk commented:
agree. we should have a separate PR for other battery methods

workingmonk commented:
looks good :+1:
